src/blop/agent.py

```diff
@@ -1139,13 +1148,18 @@ def plot_objectives(self, axes: tuple[int, int] = (0, 1), **kwargs) -> None:
         axes :
             A tuple specifying which DOFs to plot as a function of. Can be either an int or the name of DOFs.
         """
-        if len(self.dofs(active=True, read_only=False)) == 1:
-            if len(self.objectives(active=True, fitness=True)) > 0:
-                plotting._plot_fitness_objs_one_dof(self, **kwargs)
-            if len(self.objectives(active=True, constraint=True)) > 0:
-                plotting._plot_constraint_objs_one_dof(self, **kwargs)
+        self.update_models()
```

Comment:
I'm not sure we should be potentially re-training models here. I think it should be up to the user to decide if they want their models updated prior to plotting. They should really do this instead:
```python
agent.update_models()
agent.plot_objectives()
```

```suggestion
```